tests/darwin/exporter/formats/export_mask_test.py

```diff
@@ -0,0 +1,581 @@
+import pytest
+from numpy.testing import assert_array_equal
+from numpy.typing import NDArray
+from PIL import Image
+from upolygon import draw_polygon
+
+from darwin import datatypes as dt
+from darwin.exporter.formats.mask import (
+    colours_in_rle,
+    export,
+    get_or_generate_colour,
+    get_palette,
+    get_render_mode,
+    get_rgb_colours,
+    render_polygons,
+    render_raster,
+    rle_decode,
+)
+
+
+# Tests for get_palette
+def test_in_grey_mode_spreads_colors_evenly() -> None:
+    palette = get_palette("grey", ["red", "green", "blue"])
+    assert palette == {"red": 0, "green": 127, "blue": 255}
+
+    palette = get_palette("grey", ["red", "green", "blue", "yellow"])
+    assert palette == {"red": 0, "green": 85, "blue": 170, "yellow": 255}
+
+    palette = get_palette("grey", ["red", "green", "blue", "yellow", "purple"])
+    assert palette == {"red": 0, "green": 63, "blue": 127, "yellow": 191, "purple": 255}
+
+
+def test_in_index_mode_doesnt_spread_colors() -> None:
+    palette = get_palette("index", ["red", "green", "blue"])
+    assert palette == {"red": 0, "green": 1, "blue": 2}
+
+    palette = get_palette("index", ["red", "green", "blue", "yellow"])
+    assert palette == {"red": 0, "green": 1, "blue": 2, "yellow": 3}
+
+    palette = get_palette("index", ["red", "green", "blue", "yellow", "purple"])
+    assert palette == {"red": 0, "green": 1, "blue": 2, "yellow": 3, "purple": 4}
+
+
+def test_in_rgb_mode_spreads_colors() -> None:
+    palette = get_palette("rgb", ["red", "green", "blue"])
+    assert palette == {"red": 0, "green": 1, "blue": 2}
+
+    palette = get_palette("rgb", ["red", "green", "blue", "yellow"])
+    assert palette == {"red": 0, "green": 1, "blue": 2, "yellow": 3}
+
+    palette = get_palette("rgb", ["red", "green", "blue", "yellow", "purple"])
+    assert palette == {"red": 0, "green": 1, "blue": 2, "yellow": 3, "purple": 4}
+
+
+def test_get_palette_raises_value_error_when_num_categories_exceeds_maximum_for_index_mode() -> None:
+    with pytest.raises(ValueError, match="maximum number of classes supported: 254."):
+        get_palette("index", ["category"] * 255)
+
+
+def test_get_palette_raises_value_error_when_only_one_category_provided_for_grey_mode() -> None:
+    with pytest.raises(
+        ValueError, match="only having the '__background__' class is not allowed. Please add more classes."
+    ):
+        get_palette("grey", ["__background__"])
+
+
+def test_get_palette_raises_value_error_when_num_categories_exceeds_maximum_for_rgb_mode() -> None:
+    with pytest.raises(ValueError, match="maximum number of classes supported: 360."):
+        get_palette("rgb", ["category"] * 361)
+
+
+def test_get_palette_raises_value_error_when_unknown_mode_is_provided() -> None:
+    with pytest.raises(ValueError, match="Unknown mode invalid."):
+        get_palette("invalid", ["red", "green", "blue"])  # type: ignore
+
+
+# Tests for get_rgb_colours
+@pytest.mark.parametrize(
```

Comment:
Like the parametric approach and we should use it more.

Reply:
It's beautiful yep.